src/main/java/io/codelex/flightplanner/flights/FlightService.java

```diff
@@ -0,0 +1,85 @@
+package io.codelex.flightplanner.flights;
+
+import io.codelex.flightplanner.airports.AirportService;
+import io.codelex.flightplanner.api.AddFlightRequest;
+import io.codelex.flightplanner.api.SearchFlightsRequest;
+import io.codelex.flightplanner.api.SearchFlightsResponse;
+import io.codelex.flightplanner.flights.objects.Flight;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@Service
+public class FlightService {
+
+
+    private FlightRepository flightRepository;
+    private AirportService airportService;
+
+    public FlightService(FlightRepository flightRepository, AirportService airportService) {
+        this.flightRepository = flightRepository;
+        this.airportService = airportService;
+    }
+
+    public synchronized Flight saveFlight(AddFlightRequest flightRequest) {
+        try {
+            Flight flight = new Flight(flightRequest.getFrom(), flightRequest.getTo(), flightRequest.getCarrier(), flightRequest.getDepartureTime(), flightRequest.getArrivalTime());
+            if (flightRepository.getFlightList().size() > 0) {
```

Comment:
Šis IF faktiski nav nepieciešams, jo for each loops nemaz nepildīsies, ja saraksts būs tukšs.